pkg/mod/mod.go

```diff
@@ -0,0 +1,40 @@
+package mod
+
+import (
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type goModule struct {
+	Path     string     // module path
+	Version  string     // module version
+	Versions []string   // available module versions (with -versions)
+	Replace  *goModule  // replaced by this module
+	Time     *time.Time // time version was created
+	Update   *goModule  // available update, if any (with -u)
+	Main     bool       // is this the main module?
+	Indirect bool       // is this module only an indirect dependency of main module?
+	Dir      string     // directory holding files for this module, if any
+	GoMod    string     // path to go.mod file for this module, if any
+}
+
+type goModules []*goModule
```

Comment:
I'm not sure I understand the point of this type, or how it is supposed to be used.

Would it be better to make the func below `func GetByFilepath(mods goModules, path string) *goModule` which would mean the type isnt needed anymore? Do you expect users to use goModules as a slice or an object? 
`mods.Add(mod)` is nicer than `mods = append(mods, mod)`